error with upload field [emmett 2.6]

I have this model. When "archivo" of the upload type is optional. When I attach it, it works fine, when it is not attached, it gives error:

```python
class InformeReparticion(Signature):
    tablename = "informes_reparticiones"
    belongs_to({'tramite': 'Tramite'})
    belongs_to({'reparticion': 'Reparticion'})
    fecha = Field.date()
    aprobado = Field.bool()
    observaciones = Field.text()
    archivo = Field.upload()
```

> 
> [ERROR] Application callable raised an exception
> Traceback (most recent call last):
>   File "/venv/lib/python3.12/site-packages/granian/_futures.py", line 4,
>  in future_watcher
>     await inner(watcher.scope, watcher.proto)
>   File "/venv/lib/python3.12/site-packages/granian/rsgi.py", line 81, in
>  _http_logger
>     rv = await callback(scope, proto)
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/venv/lib/python3.12/site-packages/emmett_core/protocols/rsgi/ha
> ndlers.py", line 63, in __call__
>     if coro := http.rsgi(protocol):
>                ^^^^^^^^^^^^^^^^^^^
>   File "/venv/lib/python3.12/site-packages/emmett_core/http/response.py"
> , line 69, in rsgi
>     protocol.response_bytes(self.status_code, list(self.rsgi_headers), self.body)
>   File "/venv/lib/python3.12/site-packages/granian/rsgi.py", line 66, in
>  response_bytes
>     return self.inner.response_bytes(status, headers, body)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> TypeError: argument 'body': 'str' object cannot be converted to 'PyByteArray'

Sorry @gi0baro , the first message:

> [ERROR] Application callable raised an exception
> Traceback (most recent call last):
> File "/venv/lib/python3.12/site-packages/granian/_futures.py", line 4,
> in future_watcher
> await inner(watcher.scope, watcher.proto)
> File "/venv/lib/python3.12/site-packages/granian/rsgi.py", line 81, in
> _http_logger
> rv = await callback(scope, proto)
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> File "/venv/lib/python3.12/site-packages/emmett_core/protocols/rsgi/ha
> ndlers.py", line 63, in call
> if coro := http.rsgi(protocol):
> ^^^^^^^^^^^^^^^^^^^
> File "/venv/lib/python3.12/site-packages/emmett_core/http/response.py"
> , line 69, in rsgi
> protocol.response_bytes(self.status_code, list(self.rsgi_headers), self.body)
> File "/venv/lib/python3.12/site-packages/granian/rsgi.py", line 66, in
> response_bytes
> return self.inner.response_bytes(status, headers, body)
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> TypeError: argument 'body': 'str' object cannot be converted to 'PyByteArray'


corresponds to  emmett-core 1.0.2 and the second error: **Invalid multipart data** corresponds to emmett-core 1.0.3

--- a/emmett/forms.py
+++ b/emmett/forms.py
@@ -384,8 +384,8 @@ class ModelForm(BaseForm):
             #: handle uploads
             for field in filter(lambda f: f.type == "upload", self.writable_fields):
                 upload = self.files[field.name]
-                del_field = field.name + "__del"
-                if not upload.filename:
+                if not upload:
+                    del_field = field.name + "__del"
                     if self.input_params.get(del_field, False):
                         self.params[field.name] = self.table[field.name].default or ""
                         # TODO: do we want to physically delete file?

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -35,7 +35,7 @@ classifiers = [
 
 dependencies = [
     "click>=6.0",
-    "emmett-core[granian,rapidjson]~=1.0.2",
+    "emmett-core[granian,rapidjson]~=1.0.4",
     "emmett-pydal==17.3.1",
     "pendulum~=3.0.0",
     "pyyaml~=6.0",

